ecl2df/hook_implementations/jobs.py

```diff
@@ -1,20 +1,20 @@
 import importlib
-import os
+from pathlib import Path
 from pkg_resources import resource_filename
 
 from ert_shared.plugins.plugin_manager import hook_implementation
 from ert_shared.plugins.plugin_response import plugin_response
 
 
 def _get_jobs_from_directory(directory):
-    resource_directory = resource_filename("ecl2df", directory)
+    resource_directory = Path(resource_filename("ecl2df", directory))
 
     all_files = [
-        os.path.join(resource_directory, f)
-        for f in os.listdir(resource_directory)
-        if os.path.isfile(os.path.join(resource_directory, f))
+        resource_directory / filename
+        for filename in resource_directory.glob("*")
+        if (resource_directory / filename).exists()
     ]
-    return {os.path.basename(path): path for path in all_files}
+    return {str(Path(path).name): str(path) for path in all_files}
```

Comment:
The result of a `/` operator on a `Path` and a `string` will result in a new `Path` instance, hence no need to create another one here.